lib/react.dart

```diff
@@ -8,11 +8,23 @@ library react;
 /// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
 /// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
 abstract class Component {
+  Map _props;
+  Map _state = {};
+  dynamic _ref;
+
   /// ReactJS `Component` props.
-  Map props;
+  Map get props => _props;
+  set props(Map value) => _props = value;
+
+  /// ReactJS `Component` state.
+  Map get state => _state;
+  set state(Map value) => _state = value;
 
-  /// Provides access to the underlying DOM representation of the [render]ed `Component`.
-  dynamic ref;
+  /// Returns the component of the specified [ref].
+  ///
+  /// Returns a [Component] if it is a Dart component, otherwise returns an "Dom component" (Dart `Element`).
```

Comment:
#nit Doc comments for getters should be noun phrases: https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases

Not sure why the comment is referencing the getter itself—doesn't really make sense.

Also, this is missing the case for JS components.

What about:
```dart
/// A function that, when called with a String ref, returns the associated component.
///
/// This function will return either:
///
/// * [Component] for Dart components
/// * `ReactElement` for JS composite components
/// * `Element` DOM components
```